### ISSUE TITLE
Implement context for sharing Pod socket/data within app

### ISSUE THREAD

**Issue by taesungh** [UCI-HyperXite/hx9-control-system#80]
As encouraged by #41 and #71, we can have a [context](https://react.dev/learn/passing-data-deeply-with-context) to simply acquire the `podSocketClient` and `podData` within different child components without needing to do prop drilling. Additionally, `usePodData` is not a shared singleton, so calling the hooks in multiple components will make multiple socket instances which is not desired.
- Create a new `PodContext` context in `src/services/PodContext.ts` which provides the same values as `usePodData`, i.e. `podData` and `podSocketClient`
- Call `usePodData` in `App` and wrap the children in `<PodContext.Provider>` to provide the values
- Update `ControlPanel` to acquire the `podSocketClient` through `useContext` rather than calling `usePodData` again.

**Comment by taesungh** [comment]
Would be nice to have this before getting too deep in #56 but waiting on #61 since hook is being hoisted there.
